### PR TITLE
Update readme with custom error chapter and modify intro

### PR DESCRIPTION
**Pull request by gobbletsit**
Update intro without completely removing the KMM part.

Add custom error chapter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +149,24 @@ try {
...
+```kotlin
+@Serializable
+@JsonApiXError(type = "person")
+data class PersonError(
+    val code: String,
+    val title: String
+) : Error
+```
+
+In this example, the annotation processor will automatically make the error type of a `Person` class to be a `PersonError` and use it as a type when deserializing errors array. Developer needs to make sure that the `type` parameter value in `JsonApiXError` matches the one in the `JsonApiX` above the original model.
```
Comment (stjepanbanek) [inline README.md:168]: Looks good, but I would add a  small code example with the new generic extension `asJsonXHttpException<PersonError>()`

**Review by stjepanbanek — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gobbletsit) [reply README.md:168]: Agreed 👍

File: README.md [hunk 2]
```diff
@@ -100,6 +100,9 @@ listAdapter.convertToString(person) // Produces JSON API String from a Person li
 listAdapter.convertFromString(inputJsonString) // Produces Person list from JSON API String
 ```
 
+In some cases, a need for custom modification of certain aspects of model classes is preferred. With this in mind, we've enabled the option for clients to customize 
+JsonApiModel if such a need occurs. Clients are able to set optional params on a model class such as links or meta(explained in its own chapter), but also to customize the default `id` of a model. Clients can simply set/get `id` on a JsonApiModel depending on the case at hand.
+
```
Comment (stjepanbanek) [inline README.md:105]: There is a whole section related to `JsonApiModel`. I would move this there. Also use the ticks around "JsonApiModel" to make it look like code -> `JsonApiModel`

File: README.md [hunk 3]
```diff
@@ -125,23 +128,65 @@ A single error is modeled to wrap the most common arguments of an error. Develop
...
     val detail: String,
     val status: String
 )
 ```
 
-When using Retrofit, in the event of a network error a `HttpException` will be thrown. To extract the `Error` model from a response, you can
-use the `HttpException.asJsonXHttpException()` extension, which will return a `JsonXHttpException`, containing the original `response` as
-well as `errors` list.
+When using Retrofit, in the event of a network error a `HttpException` will be thrown. To extract the `DefaultError` model from a response, you can
+use the `HttpException.asJsonXHttpException<DefaultError>()` extension. The extension takes in a generic `DefaultError` which will return a `JsonXHttpException`, containing the original `response` as well as `errors` list.
```
Comment (stjepanbanek) [inline README.md:142]: `asJsonXHttpException()` doesn't have a generic parameter to get the instance of `DefaultError`. Generic version of the extension is made separately and used only for custom errors

File: README.md [hunk 4]
```diff
@@ -125,23 +128,65 @@ A single error is modeled to wrap the most common arguments of an error. Develop
...
+
+```kotlin
+try {
+    val person = io { sampleApiService.fetchPerson() }
+} catch (exception: HttpException) {
+    val errors = exception.asJsonXHttpException<DefaultError>().errors
```
Comment (stjepanbanek) [inline README.md:148]: No generic param needed here as well

File: README.md [hunk 5]
```diff
@@ -125,23 +128,65 @@ A single error is modeled to wrap the most common arguments of an error. Develop
...
+    errors?.first()?.let {
+        if (it is DefaultError) {
+            showError(it.code)
+        } else {
+            showError("Not a default error.")
+        } 
+    } ?: showError("Unable to parse to JsonXHttpException")
```
Comment (stjepanbanek) [inline README.md:155]: This code example is too specific to our sample and doesn't show the full potential of the feature in my opinion. I would keep it shorter, something like this:

```
// Using DefaultError
try {
    val person = io { sampleApiService.fetchPerson() }
} catch (exception: HttpException) {
    val jsonXHttpException = exception.asJsonXHttpException()
    val errors = jsonXHttpException.errors // List<DefaultError>
    // handle errors
    ...
}
```

File: README.md [hunk 6]
```diff
@@ -125,23 +128,65 @@ A single error is modeled to wrap the most common arguments of an error. Develop
...
     val person = io { sampleApiService.fetchPerson() }
 } catch (exception: HttpException) {
-    val errors = exception.asJsonXHttpException().errors
+    val errors = exception.asJsonXHttpException<PersonalError>().errors
+    errors?.first()?.let {
+        when (it) {
+            is DefaultError -> showError(it.code)
+            is PersonalError -> showError(it.desc)
+        }
+    } ?: showError("Unable to parse to JsonXHttpException")
```
Comment (stjepanbanek) [inline README.md:189]: Same here, I would change the example to be something like this
```
// Using custom error
try {
    val person = io { sampleApiService.fetchPerson() }
} catch (exception: HttpException) {
    val jsonXHttpException = exception.asJsonXHttpException<PersonError>()
    val errors = jsonXHttpException.errors // List<PersonError>
    // handle errors
    ...
}
```

**Review by stjepanbanek — CHANGES_REQUESTED** [review]
Some improvements can be done. Let's make this pretty for the release

File: README.md (hunk 2 above)
Comment (gobbletsit) [reply README.md:105]: As you wish, I think it's better to leave it here on customization part but will move it 👍

File: README.md (hunk 3 above)
Comment (gobbletsit) [reply README.md:142]: Didn't believe you on this so had to check, will update. :)

**Review by stjepanbanek — APPROVED** [review]

**Pull request merged**